docs: Adds details to attend issues #295, #287, #221, #199

JavierMtzRdz opened the pull request:
### Checklist

Please:

- [x] Make sure this PR is against "dev", not "main" (unless this is a release
      PR).
- [x] Request a review from one of the current epidatr main reviewers:
      brookslogan, dshemetov, nmdefries, dsweber2.
- [x] Makes sure to bump the version number in `DESCRIPTION`. Always increment
      the patch version number (the third number), unless you are making a
      release PR from dev to main, in which case increment the minor version
      number (the second number).
- [x] Describe changes made in NEWS.md, making sure breaking changes
      (backwards-incompatible changes to the documented interface) are noted.
      Collect the changes under the next release number (e.g. if you are on
      1.7.2, then write your changes under the 1.8 heading).

### Change explanations for reviewer
* Issue #295: Description for `save_api_key` is inaccurate.
- Change: Provide a clearer explanation as in the message linked to the function. 

* Issue #287: The `pvt_meta_norostat` `locations` parameter lacks valid inputs.
- Change: Provides a clearer explanation. 

* Issue #221: Reduce duplication
- Change: Adds a .epidatr_shared_params list of parameters that includes shared parameters and inherits shared parameters in the specific functions' documentation. 

* Issue #199: Group `as_of`, `issues`, and `lag`
- Change: Adds a Data Versioning section to clarify its use. 

### Magic GitHub syntax to mark associated Issue(s) as resolved when this is merged into the default branch

- Resolves #295, resolves #287, resolves #221, resolves #199

JavierMtzRdz committed:
docs: Correct `save_api_key` process
It clarifies that the API key needs to be manually updated and closes issue 295.

JavierMtzRdz committed:
doc: Indicates that the norovirus endpoint is currently inactive.
This closes issue 287 since the location does not need to be specified.

JavierMtzRdz committed:
doc: create .epidatr_shared_params base documentation and add a Data Versioning section
To reduce repetition, epidatr_shared_params contains shared parameters that are inherited by the functions’ documentation. Data Versioning is also inherited where necessary.

JavierMtzRdz committed:
docs: document (GHA)

JavierMtzRdz committed:
style: styler (GHA)

JavierMtzRdz committed:
fix: Add @keywords internal to .epidatr_shared_params
This change prevents issues with referencing .epidatr_shared_params when building the documentation site.

JavierMtzRdz committed:
docs: document (GHA)

JavierMtzRdz committed:
docs: point out valid lists of locations.
This closes essue #287.

JavierMtzRdz committed:
docs: document (GHA)

dshemetov commented on R/endpoints.R:
```diff
@@ -1,3 +1,46 @@
+#' @param auth string. Your restricted access key (not the same as API key).
+#' @param locations character. Vector of locations to fetch.
+#' @param states character. Two letter state abbreviations.
+#' @param regions character. Vector of regions to fetch.
```
praise: Thanks for taking these issues on!

question: Does R intelligently notice when a "shared parameter" isn't present in a function and leave it off? If yes, then consolidating all params up here sounds great. If not, then will this list arguments that don't apply to various endpoints? For instance, `pub_covidcast` doesn't have `regions` or `states` or `locations`, do those still show up? It might not be a big deal if there are extra arguments, just wondering what our options are for this sort of thing.

JavierMtzRdz replied:
answer: Yes, ["@inheritParams only inherits docs for arguments that the function actually uses and that aren’t already documented"](https://r-pkgs.org/man.html#:~:text=@inheritParams%20only%20inherits%20docs%20for%20arguments%20that%20the%20function%20actually%20uses%20and%20that%20aren’t%20already%20documented). Also, there are a few cases where `epiweeks` or `regions` have a different description. In those cases, roxygen2 overrides the inherited parameter description with the provided description for that function. 

Also, I reviewed the documentation to ensure it was working as intended, and it is. `pub_fluview_clinical ` as an example below. 
<img width="1365" height="716" alt="Screenshot 2025-11-05 at 12 42 25 PM" src="https://github.com/user-attachments/assets/f87def48-6798-47d4-a6ac-6ec179ca40cc" />

brookslogan replied:
issue: for fluview_clinical regions, I don't see the link to the full region list.

suggestion: make fluview and fluview_clinical `regions` description more specific.  E.g., I think something like
```r
#' @param regions character. Vector of location IDs to fetch.  Can be
#'   "nat" for national, "hhs1"--"hhs10" for HHS Regions, "cen1"--"cen9" for census divisions,
#'   lowercase two-letter state or territory abbreviations for most states and territories,
#'   "jfk" for New York City, or "ny_minus_jfk" for upstate New York.
#'   Full list of locations is available <here>.
```
w/ an actual link.

JavierMtzRdz replied:
fixed: I incorporated the detailed description you proposed. Regarding the link, I wanted to create a section within the Epidata API Documentation website, but while that is being arranged, I linked it to the [regions' labels](https://github.com/cmu-delphi/delphi-epidata/blob/main/labels/regions.txt) in the repository.

brookslogan commented on R/endpoints.R:
```diff
@@ -1,3 +1,47 @@
+#' @param locations character. Vector of locations to fetch.
+#' @param states character. Two letter state abbreviations.
+#' @param regions character. Vector of regions to fetch.
+#' @param epiweeks [`timeset`]. Epiweeks to fetch. Supports
```
todo: fix content that got cut off here

JavierMtzRdz replied:
Fixed!

JavierMtzRdz committed:
docs: It adds conditions to run the examples to replace `\dontrun`
It solves #208.

JavierMtzRdz committed:
docs: fix epiweeks that got cut off

JavierMtzRdz committed:
docs: correcting regions' description for pub_fluview functions

JavierMtzRdz committed:
docs: document (GHA)

JavierMtzRdz committed:
style: styler (GHA)

JavierMtzRdz commented:
I added changes to address issues #208 and #288. 

### Change explanations for reviewer
* Issue #208: `\dontrun` does not permit examples showing request outputs on the documentation site. It suggests using `\donttest`.
- Change: I added conditional examples with `@examplesIf` so they only run when there is internet access and the corresponding API key is available. I did not use `\donttest` as [it is checked by CRAN occasionally](https://stackoverflow.com/questions/68935126/when-and-when-not-to-use-donttest-and-dontrun-in-r-package-development#:~:text=CRAN%20will%20test%20the%20\donttest%20code%2C%20so%20it%20has%20to%20work). The examples on the documentation website appear when the API key is available. Examples with lengthy outputs are enclosed within `\dontrun` safeguards. 


* Issue #288: Audit all endpoint examples.
- Note: All examples worked. The only relevant comment is that the `pub_flusurv` example returns data with a warning.
> Warning message:
Not all return columns are specified as expected epidata fields
ℹ Unspecified fields season, rate_age_5, rate_age_6, rate_age_7, rate_age_18t29,
  rate_age_30t39, rate_age_40t49, rate_age_5t11, rate_age_12t17, rate_age_lt18,
  rate_age_gte18, rate_age_1t4, rate_age_gte75, rate_age_0tlt1, rate_race_white,
  rate_race_black, rate_race_hisp, rate_race_asian, …, rate_flu_a, and rate_flu_b may
  need to be manually converted to more appropriate classes 

### Magic GitHub syntax to mark associated Issue(s) as resolved when this is merged into the default branch

- Resolves #208, resolves #288

dshemetov commented:
Request: the Data Versioning section is good, but can we also [add two more examples to the Getting Started vignette](https://cmu-delphi.github.io/epidatr/articles/epidatr.html#getting-versioned-data) to address my suggestion in #199? Specifically of using the issues field and the lag field.

JavierMtzRdz committed:
docs: include examples using the `issues` and `lag` arguments in the “Getting Started” vignette and add references to the versioned data vignette in the versioned data documentation section.

JavierMtzRdz commented:
In the last commit, I included examples using the issues and lag arguments in the “Getting Started” vignette and added references to the versioned data vignette in the versioned data documentation section.

JavierMtzRdz committed:
docs: document (GHA)

nmdefries committed:
Merge branch 'dev' into adj-doc-issue

nmdefries committed:
news and version

nmdefries commented on R/endpoints.R:
```diff
@@ -1,22 +1,63 @@
+#' The following parameters control this and are mutually exclusive (only
+#' one can be provided at a time).
+#' \itemize{
+#'   \item \code{as_of}: (Date) Retrieve the data as it was on this date.
```
Check format. `as_of` should be able to have same format (int, epirange, etc) as other date fields

nmdefries committed:
small wording changes to params and rebuild docs

nmdefries reviewed (approved):
This looks good to me. I made some small changes to the shared params wording, and updated `DESCRIPTION` and `NEWS`.

The examples are now failing, although I didn't change them. Potentially a dependency bump?

JavierMtzRdz committed:
fix: add missing import of httr in examples.

JavierMtzRdz commented:
There was an example missing a function namespace. It has been fixed.
